### PR TITLE
Feature #1402 config cleanup model applications

### LINKED ISSUE #1402: Clean up METplus use case configuration files
*Replace italics below with details for this issue.*

## Describe the Enhancement ##
*Provide a description of the enhancement request here.*

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2792541

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
This PR involves cleaning up the METplus config files in the repository under parm/use_cases/model_applications.

I also added support for setting match_points in TCPairs wrapper because it was being set via TC_PAIRS_MET_CONFIG_OVERRIDES in a few use cases.

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Ran existing use cases and confirmed there were no differences

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Ensure all automated tests pass
Review config files to ensure organization of variables is satisfactory and all URLs to documentation are correct

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **8/2/2022**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByInit.conf [hunk 1]
```diff
@@ -1,70 +1,125 @@
...
-TC_PAIRS_BASIN =
 
-# Cyclone, a space or comma-separated list of cyclone numbers. If left empty, all cyclones will be used.
-TC_PAIRS_CYCLONE =
+###
+# TCPairs Settings
+# https://metplus.readthedocs.io/en/latest/Users_Guide/wrappers.html#tcpairs
+###
+
+TC_PAIRS_INIT_INCLUDE =
```
Comment (j-opatz) [inline parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByInit.conf:112]: For this use case's config file, there's a lot of empty (non-set) configurable options. This seems to be at odds with how the other use cases have been handled, with most non-set options being removed. If this use case is relying on the defaults for all of these settings, should they just be removed, like the other use cases?

Comment (j-opatz) [inline parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByInit.conf:112]: Should these options (e.g. TC_PAIRS_INIT_INCLUDE, TC_PAIRS_VALID_BEG, etc.) be removed if they're just being set to default (aka empty)? That seems to be more in line with how the other use cases are being handled.

File: parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByInit.conf [hunk 2]
```diff
@@ -106,15 +146,12 @@ TC_STAT_BASIN =
 TC_STAT_CYCLONE =
 TC_STAT_STORM_NAME =
 
-# Stratify by init times via a comma-separate list of init times to
-# include or exclude.  Time format defined as YYYYMMDD_HH or YYYYMMDD_HHmmss
 TC_STAT_INIT_BEG =
```
Comment (j-opatz) [inline parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByInit.conf:149]: Many of the TC_STAT options are empty here as well; seems to be common for the feature relative use cases. Should these be removed for ease of reading, or are they popular/integral to users and should stay?

**Review by j-opatz — APPROVED** [review]
I left two comments related to the feature relative cases, but they are not necessary changes to my approval of the PR. All of the links that I tested worked, and the rearrangement of the use cases, while a vast undertaking, helps create a natural order to the config files that will hopefully help users dive into these use cases easier.

**Comment by georgemccabe** [comment]
@j-opatz , thank you for bringing up the variables with empty values. While going through these changes, I removed many of the empty string values to discover any variables that are required when they should be optional. If any were found, I would update the wrapper to no longer require these variables. It appears I may have missed these few variables, so it would be good to remove them and test to ensure that the same result occurs. I can do this in a separate branch.

This can get tricky, as there are some MET config variables where the default is a list with a single value and an empty list is a valid value for that variable. In this case, the behavior will differ if the METplus variable is unset vs. if it is set to an empty string, so removing the variable would change the results by design. This is unlikely the case for these variables.

**Pull request merged**